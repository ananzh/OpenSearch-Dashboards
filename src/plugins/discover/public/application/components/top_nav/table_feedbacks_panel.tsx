/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import {
  EuiModal,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiModalBody,
  EuiModalFooter,
  EuiButton,
  EuiText,
  EuiButtonEmpty,
} from '@elastic/eui';

interface TableFeedbacksPanelProps {
  onClose: () => void;
  onTurnOff: () => Promise<void>;
}

export const TableFeedbacksPanel = ({ onClose, onTurnOff }: TableFeedbacksPanelProps) => (
  <EuiModal onClose={onClose} maxWidth={600}>
    <EuiModalHeader>
      <EuiModalHeaderTitle>
        <h4>Share your thoughts on the latest Discover features</h4>
      </EuiModalHeaderTitle>
    </EuiModalHeader>

    <EuiModalBody>
      <EuiText>
        <p>
          Event tables: Documents are now expanded through a flyout. Density, column order, and
          sorting controls have been improved.{' '}
          <a href="https://survey.opensearch.org" target="_blank" rel="noopener noreferrer">
            Provide feedbacks
          </a>
        </p>
      </EuiText>
    </EuiModalBody>

    <EuiModalFooter>
      <EuiButtonEmpty onClick={onClose}>Cancel</EuiButtonEmpty>
      <EuiButton onClick={onTurnOff} fill>
        Turn off new features
      </EuiButton>
    </EuiModalFooter>
  </EuiModal>
);
