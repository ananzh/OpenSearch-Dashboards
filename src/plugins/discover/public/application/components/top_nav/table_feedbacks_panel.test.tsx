/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { TableFeedbacksPanel } from './table_feedbacks_panel';

describe('TableFeedbacksPanel', () => {
  let onCloseMock: jest.Mock;
  let onTurnOffMock: jest.Mock;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    onCloseMock = jest.fn();
    onTurnOffMock = jest.fn();
    wrapper = shallow(<TableFeedbacksPanel onClose={onCloseMock} onTurnOff={onTurnOffMock} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onClose when the Cancel button is clicked', () => {
    wrapper.find('EuiButtonEmpty').simulate('click');
    expect(onCloseMock).toHaveBeenCalled();
  });

  it('calls onTurnOff when the Turn off new features button is clicked', () => {
    wrapper.find('EuiButton').simulate('click');
    expect(onTurnOffMock).toHaveBeenCalled();
  });
});
