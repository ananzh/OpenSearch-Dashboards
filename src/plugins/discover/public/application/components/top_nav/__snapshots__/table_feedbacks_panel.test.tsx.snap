// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFeedbacksPanel renders correctly 1`] = `
<EuiModal
  maxWidth={600}
  onClose={[MockFunction]}
>
  <EuiModalHeader>
    <EuiModalHeaderTitle>
      <h4>
        Share your thoughts on the latest Discover features
      </h4>
    </EuiModalHeaderTitle>
  </EuiModalHeader>
  <EuiModalBody>
    <EuiText>
      <p>
        Event tables: Documents are now expanded through a flyout. Density, column order, and sorting controls have been improved. 
        <a
          href="https://survey.opensearch.org"
        >
          Provide feedbacks
        </a>
        .
      </p>
    </EuiText>
  </EuiModalBody>
  <EuiModalFooter>
    <EuiButtonEmpty
      onClick={[MockFunction]}
    >
      Cancel
    </EuiButtonEmpty>
    <EuiButton
      fill={true}
      onClick={[MockFunction]}
    >
      Turn off new features
    </EuiButton>
  </EuiModalFooter>
</EuiModal>
`;
